pypsa/descriptors.py

```diff
@@ -60,31 +60,22 @@ def get_switchable_as_dense(
     --------
     >>> get_switchable_as_dense(n, 'Generator', 'p_max_pu')
     """
-    static = n.static(component)
-    dynamic = n.dynamic(component)
-
-    index = static.index
-
-    varying_i = dynamic[attr].columns
-    fixed_i = static.index.difference(varying_i)
-
-    if inds is not None:
-        index = index.intersection(inds)
-        varying_i = varying_i.intersection(inds)
-        fixed_i = fixed_i.intersection(inds)
     if snapshots is None:
         snapshots = n.snapshots
 
-    vals = np.repeat([static.loc[fixed_i, attr].values], len(snapshots), axis=0)
-    static = pd.DataFrame(vals, index=snapshots, columns=fixed_i)
-    varying = dynamic[attr].loc[snapshots, varying_i]
+    static = n.static(component)[attr]
+    empty = pd.DataFrame(index=snapshots)
+    dynamic = n.dynamic(component).get(attr, empty).loc[snapshots]
```

Comment:
If the component is defined by pypsa defaults or the user, then this returns an empty dataframe if no components are assigned. When a component is not existent in the sense that is not registered, this should raise an error, I would say. Otherwise typos and unintentionally unregistered components will not lead to the expected behaviour.